File: sejin/level3/압축.py
def solution(msg):
    set = {chr(i): i+1-ord('A') for i in range(ord('A'), ord('Z')+1)}
    answer = []
    a=0
    b=0
    while b < len(msg):
        if msg[a:b+1] in set:
            b += 1
        else:
            set[msg[a:b+1]] = len(set)+1
            answer.append(set[msg[a:b]])
            a = b
            b = b
    answer.append(set[msg[a:b+1]])
    return answer

File: qkfcl9987/20200419/압축.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

vector<int> solution(string msg) {
    vector<int> answer;
    map<string, int> map;
    int count = 26;
    for (int i = 65; i < 91; i++)
    {
        char ctr = i;
        string s(1,ctr);
        map.insert(make_pair(s,i - 64));
    }

    for (int i = 0; i < msg.size(); i++) 
    {
        char ctr1 = msg[i];
        string s1(1, ctr1);
        
        for (int j = i+1; j < msg.size(); j++) 
        {
            char ctr2 = msg[j];
            string s2(1, ctr2);
            s2 = s1 + s2;
            if (map.find(s2) == map.end())
            {
                i = j - 1;
                break;
            }
            else 
            {
                s1 = s2;
                if (j == msg.size() - 1) 
                {
                    i = j;
                    break;
                } 
            }
        }

        if (i < msg.size() - 1)
        {
            char ctr2 = msg[i + 1];
            string s2(1, ctr2);
            s2 = s1 + s2;
            map.insert(make_pair(s2, count + 1));
            count++;
        }

        answer.push_back(map.find(s1)->second);
    }

    return answer;
}